src/git/api.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+export { API, Repository, InputBox, RepositoryState, RepositoryUIState, Commit, Branch, Git, RefType, UpstreamRef, GitErrorCodes } from '../typings/git';
+import { API, Repository, Git } from '../typings/git';
+import { LiveShare } from 'vsls/vscode.js';
+import { VSLSGuest } from './vslsguest';
+import { VSLSHost } from './vslshost';
+import { Model } from './model';
+
+/**
+ * Should be removed once we fix the webpack bundling issue.
+ */
+async function getVSLSApi() {
+	const liveshareExtension = vscode.extensions.getExtension('ms-vsliveshare.vsliveshare');
+	if (!liveshareExtension) {
+		// The extension is not installed.
+		return null;
+	}
+	const extensionApi = liveshareExtension.isActive ?
+		liveshareExtension.exports : await liveshareExtension.activate();
+	if (!extensionApi) {
+		// The extensibility API is not enabled.
+		return null;
+	}
+	const liveShareApiVersion = liveshareExtension.packageJSON.version;
+	// Support deprecated function name to preserve compatibility with older versions of VSLS.
+	if (!extensionApi.getApi) {
+		return extensionApi.getApiAsync(liveShareApiVersion);
+	}
+	return extensionApi.getApi(liveShareApiVersion);
+}
+
+class CommonGitAPI implements API, vscode.Disposable {
+	git: Git;
+	private _onDidOpenRepository = new vscode.EventEmitter<Repository>();

```

Comment:
is it possible to more directly expose the api events instead of creating another layer of emitters?